### PR TITLE
Skip slow tests for non-host targets that don't depend on target features

### PR DESCRIPTION
**Pull request by dsharletg**

### CODE REVIEW

**Review by steven-johnson — APPROVED** [review]

**Comment by dsharletg** [comment]
It looks like this is skipping these tests even when HL_JIT_TARGET=host :(. It seems like maybe Target::operator!= is doing something surprising. But also maybe this is not a good way to do this.

**Comment by steven-johnson** [comment]
> It looks like this is skipping these tests even when HL_JIT_TARGET=host :(. It seems like maybe Target::operator!= is doing something surprising. But also maybe this is not a good way to do this.

That's weird. Try logging what each Target is?

File: test/correctness/strict_float.cpp [hunk 1]
```diff
@@ -257,6 +257,11 @@ Buffer<float> block_transposed_by_n(Buffer<float> &buf, int vectorize) {
 }
 
 int main(int argc, char **argv) {
+    if (get_jit_target_from_environment() != get_host_target()) {
```
Comment (abadams) [inline test/correctness/strict_float.cpp:260]: Are you sure about this one? It looks like it tests strict_float on a variety of sse/avx levels where llvm may do different types of fusion.

**Comment by dsharletg** [comment]
This probably isn't a big enough improvement to be worth debugging the issues with it.

**Pull request closed without merging**